src/api/app/models/cloud/upload_job.rb

```diff
@@ -0,0 +1,58 @@
+module Cloud
+  class UploadJob
+    include ActiveModel::Validations
+
+    attr_reader :user, :params, :user_upload_job, :backend_upload_job, :id
+    validate :validate_jobs
+
+    def initialize(user, params)
+      self.user = user
+      self.params = params
+    end
+
+    def create
+      begin
+        result = ::Backend::Api::Cloud.upload(
+         params[:project],
+         params[:package],
+         params[:repository],
+         params[:architecture],
+         params[:filename],
+         user,
+         params[:region]
+        )
+
+        self.backend_upload_job = Cloud::Backend::UploadJob.new(result)
+      rescue ActiveXML::Transport::Error, Timeout::Error => exception
+        self.backend_upload_job = Cloud::Backend::UploadJob.new(exception.message)
+        return self
+      end
+
+      result = Xmlhash.parse(result)
+      self.user_upload_job = @user.upload_jobs.create(
+       job_id: result['name'],
+       user: @user
+      )
+      self.id = result['name']
+      self
+    end
+
+    private
+
+    attr_writer :user, :params, :user_upload_job, :backend_upload_job, :id
+
+    def validate_jobs
+      if backend_upload_job.present? && !backend_upload_job.valid?
+        backend_upload_job.errors.full_messages.each do |msg|
+          errors.add(:base, msg)
```

Comment:
Should this be `errors.add(:backend_upload_job, msg)` so that we can distinguish between errors from the backend and errors from the frontend?